### PR TITLE
Add some status codes to various responses and make some minor cosmet…

### PR DESCRIPTION
**Pull request by nickcharles**
…ic changes, such as adding semicolons etc.

Diff shows basically everything, nothing should be all that surprising. Basically just made it more consistent across files. I have a bunch of questions that are comments on the actual code.

Thanks @matthewgrossman . Expect one for the models in just a little bit ;)

Edit: Just realized that @kenzshelley had a hand in a good amount of this, adding you on so you can comment if you'd like.

### CODE REVIEW

File: controllers/reservations.js [hunk 1]
```diff
@@ -16,26 +24,16 @@ router.post('/', function(req, res) {
...
   }).catch(function(error) {
     res.status(500).json({error: error});
   });
 });
 
-// Occupy spot passed in the id
+// Finish spot occupation passed in the id
 router.put("/:id/finish", function(req, res) {
   Reservation.finish(req.params.id).then(function(reservation) {
-    res.json(reservation);
```
Comment (nickcharles) [inline controllers/reservations.js]: This looks worse than it is, and it's GitHub being weird with diff. I just moved the 'get reservation' endpoint up to the top of the file.

File: controllers/users.js [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
   }).catch(function(error) {
-    res.status(500).json({error: error}); 
+    res.status(500).json({error: error});
   });
 });
 
-// Create a new User.
+// Create a new user
 router.post('/', function(req, res) {
   if (req.body.hasOwnProperty('email') && req.body.hasOwnProperty('password')) {
```
Comment (nickcharles) [inline controllers/users.js]: So is this if statement the only way to do this? Would it be possible to make it follow this structure instead:

if (doesn't have correct properties) {
   error;
   return;
}
success code here;

I've always thought that structure is a little cleaner, but obviously this is super nitpicky.

File: controllers/reservations.js [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
+    res.status(500).json({error: error});
+  });
+});
 
 // Create and reserve a new reservation
 router.post('/', function(req, res) {
   Reservation.reserve(req.body.userId, req.body.latitude, req.body.longitude)
     .then(function(reservation) {
-      res.json(reservation);
+      res.status(201).json(reservation);
```
Comment (nickcharles) [inline controllers/reservations.js]: On each of these functions we return the reservation object in JSON format, as opposed to the other files where we return the JSON of {spot: spot}. Is there a functional difference?

File: controllers/users.js [hunk 4]
```diff
@@ -28,9 +27,9 @@ router.post('/', function(req, res) {
 // Update a user
```
Comment (nickcharles) [inline controllers/users.js]: Not a criticism, just a curiosity. Both of the update functions don't do any sort of checking of the parameters sent in the request. Does this mean people can just add random fields to our objects? Or will Node not allow that.

Comment (nickcharles) [inline controllers/users.js]: Not a criticism, just a curiosity. Both of the update functions don't do any sort of checking of the parameters sent in the request. Does this mean people can just add random fields to our objects? Or will Node not allow that?

File: controllers/reservations.js (hunk 3 above)
Comment (matthewgrossman) [inline controllers/reservations.js]: I think we should return the straight up json personally.

Comment (matthewgrossman) [inline controllers/reservations.js]: Actually not sure

File: controllers/users.js (hunk 4 above)
Comment (matthewgrossman) [inline controllers/users.js]: There is currently zero checking of the fields they are sending up to us. Zero sanitizing in general

File: controllers/reservations.js (hunk 3 above)
Comment (matthewgrossman) [inline controllers/reservations.js]: I did reservation, kenz did the others. Would it be too nested if we do user : { attributes : { etc

File: controllers/users.js (hunk 4 above)
Comment (nickcharles) [inline controllers/users.js]: So people can just add any field they want, correct? Or do they have to match the fields defined in the Model?

File: controllers/reservations.js (hunk 3 above)
Comment (nickcharles) [inline controllers/reservations.js]: I think returning the straight up JSON is fine, we don't need to add an extra layer of what is effectively redundant information right?

File: controllers/users.js (hunk 4 above)
Comment (matthewgrossman) [inline controllers/users.js]: ya, the client should already know what endpoint they are hitting, so they know what schema to expect coming back

File: controllers/reservations.js (hunk 3 above)
Comment (matthewgrossman) [inline controllers/reservations.js]: ya, the client should already know what endpoint they are hitting, so they know what schema to expect coming back

Comment (nickcharles) [inline controllers/reservations.js]: Alright I'll wait for Kenz's input but I think that's a good idea.

File: controllers/users.js (hunk 2 above)
Comment (matthewgrossman) [inline controllers/users.js]: works with me. Also there is totally a good json schema checker that exists

File: controllers/reservations.js (hunk 3 above)
Comment (kenzshelley) [inline controllers/reservations.js]: Yeah I think returning just the model w/o a key is definitely fine for get/post endpoints. I do kind of like having {error: error} for errors just because it makes it more clear that it's an error since firebase error messages don't always actually include the word 'error' (status code does this too I guess, but might as well be redundant)

File: controllers/users.js (hunk 2 above)
Comment (kenzshelley) [inline controllers/users.js]: So I've had weird issues with trying to return early in express routes in the past (when working on Boozebot api -- it basically kept freaking out and claiming that I was setting the res status twice when I wasn't because I was returning), which is why I didn't do that here. I normally prefer the  'if (bad) return;' style too though, and it seems to be working fine here, so I'd say go for it

Comment (kenzshelley) [inline controllers/users.js]: (I tested it & it's working fine I mean)

File: controllers/users.js (hunk 4 above)
Comment (kenzshelley) [inline controllers/users.js]: Yeah as of now they can add any field, but we should definitely force them to match the model attributes eventually haha

**Comment by kenzshelley** [comment]
Looks good to me! One note: the default status code is 200, so you technically don't need to set it for all of the successful requests. No harm in being extra clear though, so I'll leave that up to you.

File: controllers/users.js (hunk 4 above)
Comment (nickcharles) [inline controllers/users.js]: For sure, that will be for a separate PR.

**Comment by nickcharles** [comment]
I had a feeling that was the case, wasn't positive though.

File: controllers/users.js (hunk 2 above)
Comment (nickcharles) [inline controllers/users.js]: Sweet, we can swap that in a separate PR.

File: controllers/reservations.js (hunk 3 above)
Comment (nickcharles) [inline controllers/reservations.js]: Agreed on the redundancy for errors, I'll switch up the others. How do we feel about returning the updated object on the update endpoints?
